### PR TITLE
Add php-8.0 support

### PR DESCRIPTION
**Pull request by samanthajayasinghe**

### CODE REVIEW

File: compose/php.yml [hunk 1]
```diff
@@ -405,6 +405,52 @@ services:
...
+      - totara-data:${REMOTE_DATA}
+      - $HOME/.bash_history:/root/.bash_history
+      - zsh-history:/root/.zsh_history
+      - ./shell:/root/custom_shell
+    depends_on:
+      - php-8.0-debug
+    networks:
+      - totara
+
+  php-8.0-debug:
```
Comment (derschatta) [inline compose/php.yml:426]: if xdebug is not supported yet maybe we should just leave out the container completely

File: php/php80/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,88 @@
+FROM php:8.0-rc-fpm
```
Comment (derschatta) [inline php/php80/Dockerfile:1]: shall we pin this to  `8.0.0beta3-fpm` for now and raise it manually once a new version comes out?

File: php/php80/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        libmemcached-dev \
+        zip \
+    && docker-php-ext-install -j$(nproc) zip \
+        intl \
+        soap \
+        opcache \
+        pdo_pgsql \
+        pdo_mysql \
+        pgsql \
+        mysqli \
```
Comment (derschatta) [inline php/php80/Dockerfile:31]: add `exif` here too (patch for the other versions adding this just got merged)

**Review by derschatta — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -23,6 +23,8 @@ php73 | 7.3 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/mas
 php73-debug | 7.3 + xdebug 2.9.6 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php73-debug/Dockerfile) | [![Build status PHP 7.3 Debug](https://img.shields.io/docker/build/totara/docker-dev-php73-debug.svg)](https://hub.docker.com/r/totara/docker-dev-php73-debug/)
 php74 | 7.4 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php74/Dockerfile) | [![Build status PHP 7.4](https://img.shields.io/docker/build/totara/docker-dev-php74.svg)](https://hub.docker.com/r/totara/docker-dev-php74/)
 php74-debug | 7.4 + xdebug 2.9.6 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php74-debug/Dockerfile) | [![Build status PHP 7.4 Debug](https://img.shields.io/docker/build/totara/docker-dev-php74-debug.svg)](https://hub.docker.com/r/totara/docker-dev-php74-debug/)
+php74 | 8.0 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php80/Dockerfile) | [![Build status PHP 8.0](https://img.shields.io/docker/build/totara/docker-dev-php80.svg)](https://hub.docker.com/r/totara/docker-dev-php80/)
```
[the remark this answers is not in the document]
Comment (samanthajayasinghe) [reply README.md]: Done

File: README.md [hunk 5]
```diff
@@ -23,6 +23,8 @@ php73 | 7.3 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/mas
...
+php74-debug | 8.0 + xdebug 2.9.6 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php80-debug/Dockerfile) | [![Build status PHP 8.0 Debug](https://img.shields.io/docker/build/totara/docker-dev-php80-debug.svg)](https://hub.docker.com/r/totara/docker-dev-php80-debug/)
```
[the remark this answers is not in the document]
Comment (samanthajayasinghe) [reply README.md]: xdebug is not working and removed the container

File: compose/php.yml (hunk 1 above)
Comment (samanthajayasinghe) [reply compose/php.yml]: done

File: php/php80/Dockerfile (hunk 3 above)
Comment (samanthajayasinghe) [reply php/php80/Dockerfile:31]: done

File: README.md [hunk 6]
```diff
@@ -62,7 +63,7 @@ Please check out the [contribute](CONTRIBUTE.md) page for more information on ho
 
 __Example:__
 ```bash
-127.0.0.1   localhost totara54 totara54.debug totara54.behat totara55 totara55.debug totara55.behat totara55 totara55.debug totara56.behat totara70 totara70.debug totara70.behat totara71 totara71.debug totara71.behat totara72 totara72.debug totara72.behat totara73 totara73.debug totara73.behat totara74 totara74.debug totara74.behat
+127.0.0.1   localhost totara54 totara54.debug totara54.behat totara55 totara55.debug totara55.behat totara55 totara55.debug totara56.behat totara70 totara70.debug totara70.behat totara71 totara71.debug totara71.behat totara72 totara72.debug totara72.behat totara73 totara73.debug totara73.behat totara74 totara74.debug totara74.behat totara80 totara80.debug totara80.behat
```
Comment (derschatta) [inline README.md:66]: debug is still present here

File: compose/nginx.yml [hunk 7]
```diff
@@ -45,6 +45,14 @@ services:
           - totara73.debug
           - totara73.behat
           - totara73.behat.totaralms.com
+          - totara74
+          - totara74.debug
+          - totara74.behat
+          - totara74.behat.totaralms.com
+          - totara80
+          - totara80.debug
```
Comment (derschatta) [inline compose/nginx.yml:53]: remove debug

File: nginx/config/totara.conf [hunk 8]
```diff
@@ -148,3 +148,20 @@ server {
...
     include totara-server.conf;
 }
+server {
+    server_name totara80;
+    set $upstream php-8.0;
+    include totara-server.conf;
+}
+
+server {
+    server_name totara80.debug;
```
Comment (derschatta) [inline nginx/config/totara.conf:158]: remove debug

**Review by derschatta — CHANGES_REQUESTED** [review]

**Comment by markmetcalfe** [comment]
The directory ```php/php-8.0-cron``` doesn't exist, but yet in ```compose/build.yml```, ```php-8.0-cron``` is defined. This results in not being able to build any containers locally (which is probably what caused the travis build to fail too)

File: php/php80/Dockerfile (hunk 2 above)
Comment (samanthajayasinghe) [reply php/php80/Dockerfile]: done

**Comment by samanthajayasinghe** [comment]
> The directory `php/php-8.0-cron` doesn't exist, but yet in `compose/build.yml`, `php-8.0-cron` is defined. This results in not being able to build any containers locally (which is probably what caused the travis build to fail too)

@markmetcalfe php-8.0-cron dir and DockerFile is in this  PR. just rebase the PR

File: php/php80/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,89 @@
+FROM php:8.0.0beta3-fpm
```
Comment (derschatta) [inline php/php80/Dockerfile:1]: @samanthajayasinghe can you update this to `php:8.0.0rc-fpm`?

File: README.md [hunk 10]
```diff
@@ -23,6 +23,7 @@ php73 | 7.3 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/mas
 php73-debug | 7.3 + xdebug 2.9.6 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php73-debug/Dockerfile) | [![Build status PHP 7.3 Debug](https://img.shields.io/docker/build/totara/docker-dev-php73-debug.svg)](https://hub.docker.com/r/totara/docker-dev-php73-debug/)
 php74 | 7.4 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php74/Dockerfile) | [![Build status PHP 7.4](https://img.shields.io/docker/build/totara/docker-dev-php74.svg)](https://hub.docker.com/r/totara/docker-dev-php74/)
 php74-debug | 7.4 + xdebug 2.9.6 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php74-debug/Dockerfile) | [![Build status PHP 7.4 Debug](https://img.shields.io/docker/build/totara/docker-dev-php74-debug.svg)](https://hub.docker.com/r/totara/docker-dev-php74-debug/)
+php80 | 8.0-beta3 |  [Dockerfile](https://github.com/totara/totara-docker-dev/blob/master/php/php80/Dockerfile) | [![Build status PHP 8.0](https://img.shields.io/docker/build/totara/docker-dev-php80.svg)](https://hub.docker.com/r/totara/docker-dev-php80/)
```
Comment (derschatta) [inline README.md:26]: 8.0-rc

**Review by derschatta — CHANGES_REQUESTED** [review]

File: php/php80/Dockerfile (hunk 9 above)
Comment (derschatta) [reply php/php80/Dockerfile:1]: hm that did not seem to work for me, still got beta3 after building, maybe try `php:8.0.0RC2-fpm-buster`

**Comment by derschatta** [comment]
@samanthajayasinghe could you also reword the PR and the commit messages to include "experimental support" for PHP 8? That would make it clear that this is not final / stable

File: README.md (hunk 10 above)
Comment (derschatta) [inline README.md:26]: we should update this as it's the final version now

File: compose/nginx.yml [hunk 11]
```diff
@@ -45,6 +45,13 @@ services:
           - totara73.debug
           - totara73.behat
           - totara73.behat.totaralms.com
+          - totara74
+          - totara74.debug
+          - totara74.behat
+          - totara74.behat.totaralms.com
+          - totara80
```
Comment (derschatta) [inline compose/nginx.yml:52]: I think xdebug 3.0.0 add PHP support so we can add `totara80.debug`

File: compose/php.yml [hunk 12]
```diff
@@ -405,6 +405,32 @@ services:
     networks:
       - totara
 
+  php-8.0:
```
Comment (derschatta) [inline compose/php.yml:408]: add a section for php-8.0-debug

File: nginx/config/totara.conf [hunk 13]
```diff
@@ -148,3 +148,15 @@ server {
     set $upstream php-7.4;
     include totara-server.conf;
 }
+
+server {
```
Comment (derschatta) [inline nginx/config/totara.conf:152]: add a section for totara80.debug

**Review by derschatta — CHANGES_REQUESTED** [review]
only changing the version in the readme.

let's add xdebug suppport using Xdebug 3.0.4 in a separate patch

File: nginx/config/totara.conf (hunk 13 above)
Comment (samanthajayasinghe) [reply nginx/config/totara.conf:152]: will create separate PR to fix php80 xdbug

File: compose/nginx.yml (hunk 11 above)
Comment (samanthajayasinghe) [reply compose/nginx.yml:52]: will create separate PR to fix php80 xdbug

File: README.md (hunk 10 above)
Comment (samanthajayasinghe) [reply README.md]: done

**Review by derschatta — APPROVED** [review]

**Pull request merged**